pennylane/templates/subroutines/qsvt.py

```diff
@@ -21,131 +21,203 @@
+
+            A = np.array([[-0.1, 0, 0, 0.1], [0, 0.2, 0, 0], [0, 0, -0.2, -0.2], [0.1, 0, -0.2, -0.1]])
+
+            dev = qml.device("default.qubit")
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.qsvt(A, poly, encoding_wires=[0, 1, 2, 3, 4], block_encoding="fable")
+                return qml.state()
+
+            matrix = qml.matrix(circuit, wire_order=[0, 1, 2, 3, 4])()
+
+        .. code-block:: pycon
+
+            >>> print(np.round(matrix[:4, :4], 4).real)
+            [[-0.0954  0.     -0.0056 -0.0054]
+             [ 0.     -0.0912 -0.     -0.    ]
+             [-0.0056  0.     -0.0788  0.0164]
+             [-0.0054 -0.      0.0164 -0.0842]]
+    """
+
+    angles = qml.math.poly_to_angles(poly, "QSVT")
     projectors = []
 
-    if convention == "Wx":
-        angles = _qsp_to_qsvt(angles)
-        global_phase = (len(angles) - 1) % 4
+    # If the input A is a Hamiltonian
+    if hasattr(A, "pauli_rep"):
 
-        if global_phase:
-            with qml.QueuingManager.stop_recording():
-                global_phase_op = qml.GlobalPhase(-0.5 * np.pi * (4 - global_phase), wires=wires)
+        if block_encoding not in ["prepselprep", "qubitization", None]:
+            raise ValueError("block_encoding should take the value 'prepselprep' or 'qubitization'")
 
-    for idx, phi in enumerate(angles):
-        dim = c if idx % 2 else r
-        with qml.QueuingManager.stop_recording():
-            projectors.append(PCPhase(phi, dim=dim, wires=wires))
+        if block_encoding == "qubitization":
+            encoding = qml.Qubitization(A, control=encoding_wires)
 
-    projectors = projectors[::-1]  # reverse order to match equation
+        else:
+            encoding = qml.PrepSelPrep(A, control=encoding_wires)
 
-    if convention == "Wx" and global_phase:
-        return qml.prod(global_phase_op, QSVT(UA, projectors))
-    return QSVT(UA, projectors)
+        projectors = [
+            qml.PCPhase(angles[i], dim=2 ** len(A.wires), wires=encoding_wires + A.wires)
+            for i in range(len(angles))
+        ]
+
+    else:
+
+        if block_encoding not in ["embedding", "fable", None]:
+            raise ValueError("block_encoding should take the value 'embedding' or 'fable'")
+
+        if qml.math.shape(A) == () or qml.math.shape(A) == (1,):
+            A = qml.math.reshape(A, [1, 1])
+
+        A = qml.math.array(A)
+
+        if block_encoding == "embedding":
+            c, r = qml.math.shape(A)
+
+            for idx, phi in enumerate(angles):
+                dim = c if idx % 2 else r
+                projectors.append(qml.PCPhase(phi, dim=dim, wires=encoding_wires))
+
+            encoding = qml.BlockEncode(A, wires=encoding_wires)
+
+        else:
+
+            # It is normalized to ensure that the block encoding is the desired one.
+            s = int(np.ceil(np.log2(max(len(A), len(A[0])))))
+            encoding = qml.FABLE(2**s * A, wires=encoding_wires)
```

Comment:
I extracted from the last cell in :

https://docs.pennylane.ai/en/stable/code/api/pennylane.FABLE.html

Basically FABLE does block encoding of A/2^n, which is not very intuitive since what we want is to do block encoding of A. Therefore, I am doing FABLE(2^n*A), so that the final result is block encoding of A.

Reply:
Ah I see, I would then recommend something like my suggestion below.